src/models/room.js

```diff
@@ -1352,7 +1356,10 @@ Room.prototype._revertRedactionLocalEcho = function(redactionEvent) {
         // re-render after undoing redaction
         this.emit("Room.redactionCancelled", redactionEvent, this);
         // reapply relation now redaction failed
-        if (redactedEvent.isRelation()) {
+        if (
+            redactedEvent.isRelation() &&
+            redactedEvent.status !== EventStatus.CANCELLED
```

Comment:
Maybe some deeper part of the aggregation path should always test for cancelled events, instead of sprinkling this check here?